main.py

```diff
@@ -1,13 +1,21 @@
 import argparse
 import sys
 from stix_shifter import stix_shifter
+import json
 
 
 def __main__():
     """
     In the case of converting a stix pattern to datasource query, arguments will take the form of...
-    <module> <translate_type> <data>
+    <module> <translate_type> <data> <options>
     The module and translate_type will determine what module and method gets called
+    Options argument comes in as:
+      "{
+          "select_fields": <string array of fields in the datasource select statement>},
+          "mapping": <mapping hash for either stix pattern to datasource or data results to stix observation objects>,
+          "result_limit": <integer to limit number or results in the data source query>,
+          "time_range": <time window (ie. last 5 minutes) used in the data source query when START STOP qualifiers are absent>
```

Comment:
What are you accepting for a time_range value? An integer?

Reply:
No, a string. The assumption is that when you are translating to the datasource query, you know what format the time range should look like. Hence the example of "last 5 minutes"